wandb/sdk/wandb_run.py

```diff
@@ -2039,6 +2057,35 @@ def close(self) -> None:
             self.lock.release()
 
 
+class CRDedupedFile(WriteSerializingFile):
```

Comment:
CRDedupedFile seems out of place. can you find somewhere else to put this?
maybe in lib/something
along with WriteSerializingFile maybe?